### ISSUE TITLE
How run paperless without encryption

### ISSUE THREAD

**Issue by erikarvstedt** [erikarvstedt/unencrypted-paperless#1]
1. Use the Paperless version from [this branch](https://github.com/erikarvstedt/paperless/tree/ea-disable-encryption)
2. When running paperless, set the environment variable `PAPERLESS_STORAGE_TYPE=unencrypted`

**Comment by GarethFox** [comment]
SupaGreat!
Plus your branch also has 'Account for KeyError problem in #345' patched!
I will give it a try asap and get back to you soon once the docker stack running using your environment variable.
Thanks a lot allowing us to access this feature!

**Comment by GarethFox** [comment]
Hi @erikarvstedt !
I am about to build a paperless stack based on your branch, I will add the environment variable 
`PAPERLESS_STORAGE_TYPE=unencrypted` in the `docker-compose.env` file as mentioned. Just one thing unclear, shall I define or delete the `PAPERLESS_PASSPHRASE=CHANGE_ME`? What would be the impact to define or remove that one regarding this new variable you set in your branch?
Thank you in advance,

**Comment by GarethFox** [comment]
For the moment, whatever I comment out or define the `PAPERLESS_PASSPHRASE=CHANGE_ME` variable I have a error message trying to build up the app using `docker-compose`
```
ERROR: for paper-back-ep  Container "c018c63a8bfc" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```

**Comment by erikarvstedt** [comment]
If you want to create a new database with encryption disabled, just define `PAPERLESS_STORAGE_TYPE=unencrypted` in `docker-compose.env`.
`PAPERLESS_PASSPHRASE` is only needed if there are still encrypted docs in your database.

To get rid of the error message, try deleting the container (`docker kill c018c63a8bfc; docker rm c018c63a8bfc`) and re-run `docker compose up`.

**Comment by GarethFox** [comment]
okay, I tried again building from your branch from scratch within my docker-compose stack.
I reuse a configuration working in an other server currently based on master branch.
Here is what I use for building your repo.
`docker-compose.env`:
```
PAPERLESS_STORAGE_TYPE=unencrypted
PAPERLESS_OCR_LANGUAGES=fra deu eng
DJANGO_LOG_LEVEL=DEBUG
```
`docker-compose.yml`
```
version: '2.1'

services:
# PUBLIC web proxy
  nginx-proxy:
    image: jwilder/nginx-proxy
    container_name: nginx-proxy
    restart: unless-stopped
    ports:
      - 80:80
      - 443:443
    volumes:
      - /docker/nginx/vhost.d:/etc/nginx/vhost.d
      - /docker/nginx/conf.d:/etc/nginx/conf.d
      - /docker/nginx/html:/usr/share/nginx/html
      - /docker/nginx/certs:/etc/nginx/certs:ro
      - /var/run/docker.sock:/tmp/docker.sock:ro
    networks:
      - frontend

# PUBLIC cert manager
  nginx-letsencrypt:
    image: jrcs/letsencrypt-nginx-proxy-companion:latest
    container_name: nginx-letsencrypt
    restart: unless-stopped
    volumes:
      - /docker/nginx/certs:/etc/nginx/certs:rw
      - /var/run/docker.sock:/var/run/docker.sock:ro
    volumes_from:
      - nginx-proxy
    depends_on:
      - nginx-proxy
    networks:
      - frontend

# PUBLIC paper-app
  paper-app:
    build:
      context: ./paperless/
      dockerfile: ./Dockerfile
    container_name: paper-app
    restart: unless-stopped
    expose:
      - '8000'
    healthcheck:
      test: ["CMD", "curl" , "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    volumes:
      - /docker/paperless/data:/usr/src/paperless/data
      - /docker/paperless/media:/usr/src/paperless/media
    env_file: ./paperless/docker-compose.env
    environment:
      - VIRTUAL_HOST=paper.local
    command: ["runserver", "--insecure", "--noreload", "0.0.0.0:8000"]
    networks:
      - frontend
      - backend

# PRIVATE paper-back
  paper-back:
    build:
      context: ./paperless/
      dockerfile: ./Dockerfile
    container_name: paper-back
    restart: unless-stopped
    depends_on:
      paper-app:
        condition: service_healthy
    volumes:
      - /docker/paperless/data:/usr/src/paperless/data
      - /docker/paperless/media:/usr/src/paperless/media
      - /docker/paperless/consume:/consume
      - /docker/paperless/export:/export
    env_file: ./paperless/docker-compose.env
    command: ["document_consumer"]
    networks:
      - backend

networks:
  frontend:
  backend:
```
`docker-compose.yml` is in the parent folder of /paperless folder where can be find the repo sources I downloaded.

`docker kill` `docker rm` and re-up the stack does not change anything.
`docker ps -a` shows paper-app looping in restarting mode.
`docker-compose logs paper-app` returns nothing special.
I changed `command: ["runserver", "--insecure", "--noreload", "0.0.0.0:8000"]` by `command: ["/usr/src/paperless/src/manage.py", "runserver", "--insecure", "0.0.0.0:8000"]` after reading this issue https://github.com/danielquinn/paperless/issues/317
Then `docker-compose logs paper-app` returns:
```
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied
paper-app            | /sbin/docker-entrypoint.sh: line 106: /usr/src/paperless/src/manage.py: Permission denied

```
Basically I can't have an healthy paper-app, then docker can't mount the paper-back.

**Comment by erikarvstedt** [comment]
Sorry, I can't help you with your specific setup.

But let's try this:
I've created a simple docker-compose config that's guaranteed to work:
- Checkout [this branch](https://github.com/erikarvstedt/paperless/tree/docker-cfg-no-encryption).
- Run `docker-compose up` from the repo root dir.
  All user data is stored in `./run`. The consume directory is `./run/consume`.

Now try to modifiy this config to fit your particular needs.

**Comment by GarethFox** [comment]
Thanks a lot.
Seems that I can have healthy paperless stack on a minimalist environment with proxy+ssl manager, but can't have healthy webapp container on more complex stacks containing more apps+db's.
I wonder why it generates conflict in complex stacks whereas it seems to work on more simple ones...

**Comment by GarethFox** [comment]
Here is as far as I could get in the debug.
I have now a steady docker-compose.yml containing my full stack plus paperless.
Paperless part is the following: 
```
# PUBLIC paper-app
  paper-app:
    build:
      context: ./paperless/
      dockerfile: ./Dockerfile
    container_name: paper-app
    restart: unless-stopped
    expose:
      - '8000'
    healthcheck:
      test: ["CMD", "curl" , "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 10
    volumes:
      - /docker/paperless/data:/usr/src/paperless/data
      - /docker/paperless/media:/usr/src/paperless/media
    env_file: ./paperless/docker-compose.env
    environment:
      - VIRTUAL_HOST=paper.example.com
      - LETSENCRYPT_HOST=paper.example.com
      - LETSENCRYPT_EMAIL=admin@example.com
      - REUSE_PRIVATE_KEYS=true
      - PAPERLESS_OCR_LANGUAGES=
    command: ["runserver", "--insecure", "--noreload", "0.0.0.0:8000"]
    networks:
      public:
      private:
        ipv4_address: 172.200.0.16

# PRIVATE paper-back
  paper-back:
    build:
      context: ./paperless/
      dockerfile: ./Dockerfile
    container_name: paper-back
    restart: unless-stopped
    depends_on:
      paper-app:
        condition: service_healthy
    volumes:
      - /docker/paperless/data:/usr/src/paperless/data
      - /docker/paperless/media:/usr/src/paperless/media
      - /home/user/paperless/consume:/consume
      - /docker/paperless/export:/export
    env_file: ./paperless/docker-compose.env
    command: ["document_consumer"]
    networks:
      private:
        ipv4_address: 172.200.0.17
```
Then, in the folder containing the `docker-compose.yml` file, I clone the first branch you mentioned `git clone https://github.com/erikarvstedt/paperless.git`.
Result is:
```
ERROR: for paper-back  Container "475fffdc615b" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```
Then I tried with your last branch mentioned `git clone https://github.com/erikarvstedt/paperless.git -b docker-cfg-no-encryption`. (just commented `PAPERLESS_DISABLE_LOGIN=true` in `.env` file)
Result is:
```
ERROR: for paper-back  Container "7f7d47d4b540" is unhealthy.
ERROR: Encountered errors while bringing up the project.
```
Finally I redo the test cloning the original-master `git clone https://github.com/danielquinn/paperless.git`
Result is positive and all the stack is working with healthy paperless stack.

I am afraid I am not able to use your branch straight away to enjoy no encryption file storage...

**Comment by erikarvstedt** [comment]
Stop all containers.
Delete all containers.
Retry.
The config in `docker-cfg-no-encryption` is guaranteed to work.
If there are indeed errors when launching the container, post them here.

**Comment by GarethFox** [comment]
Yep,
No error lunching your stack using the original `docker-compose.yml` file

**Comment by GarethFox** [comment]
Everything went okay until I scan my first doc.
I can have it from the documents page with the thumbnail associated.
I can consult document details + OCR'ed text.
But when I click on the thumbnail to download the document I get the following error (short one):
```

Request Method: | GET
-- | --
http://paper.local/fetch/doc/1
1.11.12
FileNotFoundError
[Errno 2] No such file or directory: '/usr/src/paperless/src/../media/documents/originals/0000001.pdf'
/usr/src/paperless/src/documents/models.py in source_file, line 274
/usr/bin/python3
3.6.3
['/usr/src/paperless/src',  '/usr/lib/python36.zip',  '/usr/lib/python3.6',  '/usr/lib/python3.6/lib-dynload',  '/usr/lib/python3.6/site-packages']
Tue, 15 May 2018 15:20:28 +0000
```

This aside, I can access the pdf file from shared folder between the host and the container.

**Comment by erikarvstedt** [comment]
Are you saying that the document thumbnails are displayed correctly in the document view but the document download fails?

What's the content of `./run/data/documents`?
(Post the output of `tree ./run/data/documents`)

Are you sure the `docker-compose.yml` file that you're using is 100% identical to [this one](https://github.com/erikarvstedt/paperless/blob/docker-cfg-no-encryption/docker-compose.yml) (it's the one from branch the [docker-cfg-no-encryption](https://github.com/erikarvstedt/paperless/blob/docker-cfg-no-encryption))?

As I said, my example config works 100% correctly. You must have changed something.

**Comment by GarethFox** [comment]
Yes this is the case I have, thumbnails displayed but GET error on click to download the document.
No, it is not 100% identical because my goal is to adapt the app to a specific usage.
I wanted to have the consume folder from a remote samba mounted folder, but it is just too tricky to handle permissions on a remote folder. This is why I had a lot of troubles earlier.
Now I tried to point the consume folder to /home/user/paperless/consume and I added the environment variables UID and GID according to my user.
I also pointed a /home/user/paperless/processed folder marching /usr/src/paperless/media/documents/ to then access the host local samba folder via the network to drad/drop scans and consult unencrypted processed pdf after, documents are well consumed and moved into these folder according to this adaptation but I have an error when I want to download them from the GUI running under a proxied https access as I seek a secure login.

**Comment by erikarvstedt** [comment]
You've made an error when adapting the volumes in your `docker-compose.yml` file.
Check and fix those definitions.

**Comment by GarethFox** [comment]
I does not look like, I can have access to processed documents from actual volumes definition in `docker-compose.yml`:
```
    volumes:
      - /docker/paperless/data:/usr/src/paperless/data
      - /docker/paperless/media:/usr/src/paperless/media
      - /docker/paperless/export:/export
      - /home/user/paperless/ep/consumer:/consume
      - /home/user/paperless/ep/processed:/usr/src/paperless/media/documents/originals
[...]
    command: ["document_consumer"]

```
Consume and storage is functional

**Comment by erikarvstedt** [comment]
Have you added the same volumes config to the webserver definition?

**Comment by GarethFox** [comment]
I followed the same logic according to the original `docker-compose.yml` for the webserver:
```
    volumes:
      - /docker/paperless/data:/usr/src/paperless/data
      - /docker/paperless/media:/usr/src/paperless/media
[...]
    command: ["runserver", "--insecure", "--noreload", "0.0.0.0:8000"]

```

**Comment by GarethFox** [comment]
This is more detailed debug from `docker-compose logs -f` when trying to download the file:
```
paper-web            | [15/May/2018 16:38:28] "GET /static/admin/img/sorting-icons.svg HTTP/1.1" 200 1097
paper-web            | Internal Server Error: /fetch/doc/1
paper-web            | Traceback (most recent call last):
paper-web            |   File "/usr/lib/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
paper-web            |     response = get_response(request)
paper-web            |   File "/usr/lib/python3.6/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
paper-web            |     response = self._get_response(request)
paper-web            |   File "/usr/lib/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
paper-web            |     response = self.process_exception_by_middleware(e, request)
paper-web            |   File "/usr/lib/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
paper-web            |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
paper-web            |   File "/usr/lib/python3.6/site-packages/django/views/generic/base.py", line 68, in view
paper-web            |     return self.dispatch(request, *args, **kwargs)
paper-web            |   File "/usr/src/paperless/src/paperless/mixins.py", line 23, in dispatch
paper-web            |     request, *args, **kwargs
paper-web            |   File "/usr/lib/python3.6/site-packages/django/views/generic/base.py", line 88, in dispatch
paper-web            |     return handler(request, *args, **kwargs)
paper-web            |   File "/usr/lib/python3.6/site-packages/django/views/generic/detail.py", line 117, in get
paper-web            |     return self.render_to_response(context)
paper-web            |   File "/usr/src/paperless/src/documents/views.py", line 60, in render_to_response
paper-web            |     self._get_raw_data(self.object.source_file),
paper-web            |   File "/usr/src/paperless/src/documents/models.py", line 274, in source_file
paper-web            |     return open(self.source_path, "rb")
paper-web            | FileNotFoundError: [Errno 2] No such file or directory: '/usr/src/paperless/src/../media/documents/originals/0000001.pdf'
paper-web            | [15/May/2018 16:38:32] "GET /fetch/doc/1 HTTP/1.1" 500 92394
```
Thanks you so far for your support @erikarvstedt, you are really kind.

**Comment by GarethFox** [comment]
```
tree /docker/paperless/media/documents/
/docker/paperless/media/documents/
├── originals
└── thumbnails
    └── 0000001.png

```

**Comment by GarethFox** [comment]
```
tree /home/user/paperless/ep/
/home/user/paperless/ep/
├── consumer
└── processed
    └── 0000001.pdf

```

**Comment by GarethFox** [comment]
I was expecting to find the pdf both in `originals` and `processed` folder according to volumes' definition. In my mind they were mirrors from `/usr/src/paperless/media/documents/originals` but the output is found only in `processed` according to the trees

**Comment by erikarvstedt** [comment]
You've forgotten to add this line to the webserver volume definitions:
```
- /home/user/paperless/ep/processed:/usr/src/paperless/media/documents/originals
```

**Comment by GarethFox** [comment]
I see the point.
I thought the weserver will have it by heritage with:
`- /docker/paperless/media:/usr/src/paperless/media`
as the `media` folder contains it.
Super, I try to add this tomorrow and will let you know.
If it works the stack will be fully functional !
Thank you very much @erikarvstedt

**Comment by GarethFox** [comment]
All is fine now!
Thank you again @erikarvstedt, you rock ^^

**Comment by erikarvstedt** [comment]
Awesome! :cat:

**Comment by GarethFox** [comment]
Yes pretty awesome!
Do you have any clue what portions of the code I can have a look into in order to adapt the actual incremental filenaming of the processed documents?
I would like to name them in the same way we can have via the downloading feature from the GUI with date+original document name.
Thanks,

**Comment by erikarvstedt** [comment]
Yes, I have, but it's a very bad idea because you have to keep the file name in sync with the internal data and you must handle possible duplicates.

**Comment by GarethFox** [comment]
Okay, I got it.
I was asking in case it was easy to implement, will stick to the increment naming though!
